src/Build.UnitTests/BinaryLogger_Tests.cs

```diff
@@ -138,6 +140,31 @@ public void BinaryLoggerShouldSupportFilePathExplicitParameter()
             ObjectModelHelpers.BuildProjectExpectSuccess(s_testProject, binaryLogger);
         }
 
+        [Fact]
+        public void BinaryLoggerShouldEmbedFilesViaTaskOutput()
+        {
+            using var buildManager = new BuildManager();
+            var binaryLogger = new BinaryLogger()
+            {
+                Parameters = $"LogFile={_logFile}",
+                CollectProjectImports = BinaryLogger.ProjectImportsCollectionMode.ZipFile,
+            };
+            var testProject = @"
+<Project>
+    <Target Name=""Build"">
+        <WriteLinesToFile File=""./testtaskoutputfile.txt"" Lines=""abc;def;ghi""/>
+        <CreateItem Include=""./testtaskoutputfile.txt"">
+            <Output TaskParameter=""Include"" ItemName=""EmbedInBinlog"" />
+        </CreateItem>
+    </Target>
+</Project>";
+            ObjectModelHelpers.BuildProjectExpectSuccess(testProject, binaryLogger);
+            var projectImportsZipPath = Path.ChangeExtension(_logFile, ".ProjectImports.zip");
+            using var fileStream = new FileStream(projectImportsZipPath, FileMode.Open);
+            using var zipArchive = new ZipArchive(fileStream, ZipArchiveMode.Read);
```

Comment:
Consider using TestEnvironment and TransientZipArchive?

Reply:
TransientZipArchive is only for creating a zip archive not the other way around and i don't think TestEnvironment would help here really - `ObjectModelHelpers.BuildProjectExpectSuccess` already uses it internally, the _logFile path is attached to it implicitly - pretty much the only path i produce is the `.ProjectImports.zip`, which is just conventionally attached to the binlog path.